### ISSUE TITLE
Fewer variants found in VCF file

### ISSUE THREAD

**Issue by Zepeng-Mu** [alexdobin/STAR#992]
Hi, I am using STAR WASP mode with an unzipped VCF file containing 10 columns for >5M variants. But in the running Log of STAR I saw `Loaded VCF data, found 1809185 SNPs`.

The number of SNPs found is much smaller than the line numbers in the input VCF file. I am wondering why this is happening and whether this is normal.

Thank you so much.

**Comment by alexdobin** [comment]
Hi Zepeng,

STAR-WASP only works with heterozygous single-nucleotide variants, which should be the 1.8M found in the VCF.

Cheers
Alex

**Comment by Zepeng-Mu** [comment]
Hi, thanks for your response.
So does this mean that when I am processing a large number of samples, I need to use a different VCF file for each sample?
Or could I just set every SNP to heterozygous in one file?

Thanks again

**Comment by alexdobin** [comment]
Hi Zepeng,

if the samples are from one individual, you can use the same VCF file - and it will be filtered the same way for all samples.
The "filtering" is internal, so the VCF file is not overwritten by STAR.

Cheers
Alex

**Comment by Zepeng-Mu** [comment]
Hi, thank you for your response!
I actually need to generate one VCF file for each individual. I also saw discussions in some of the earlier issues.
I did not notice this when I was reading the documents before and used a gzipped VCF with all individuals. I am wondering whether this 10-column unzipped VCF is required by some later versions of STAR, or has been like this from the very beginning?
Also closing this issue now since my questions have been solved.
Thanks!
